File: bindings/csharp/AinosSdk/Models/AinosAuthException.cs
namespace AinosSdk.Models;

/// <summary>
/// Raised when authentication with the daemon fails.
/// </summary>
public class AinosAuthException : AinosException
{
    /// <summary>
    /// The reason for the authentication failure.
    /// </summary>
    public string? Reason { get; }

    /// <summary>
    /// Whether the session has expired and needs re-authentication.
    /// </summary>
    public bool IsSessionExpired { get; }

    public AinosAuthException()
    {
    }

    public AinosAuthException(string message) : base(message)
    {
    }

    public AinosAuthException(string message, Exception innerException) : base(message, innerException)
    {
    }

    public AinosAuthException(string message, string reason) : base(message)
    {
        Reason = reason;
    }

    public AinosAuthException(string message, bool isSessionExpired) : base(message)
    {
        IsSessionExpired = isSessionExpired;
    }

    public AinosAuthException(string message, string reason, bool isSessionExpired) : base(message)
    {
        Reason = reason;
        IsSessionExpired = isSessionExpired;
    }
}

File: bindings/java/src/main/java/com/ainos/sdk/models/AinosAuthException.java
package com.ainos.sdk.models;

/**
 * Exception thrown when authentication with the Ainos daemon fails.
 * <p>
 * This can occur when:
 * <ul>
 *   <li>The provided bearer token is invalid or expired</li>
 *   <li>Authentication is required but no token was provided</li>
 *   <li>The daemon returns an authentication error response</li>
 * </ul>
 */
public class AinosAuthException extends AinosException {

    private static final long serialVersionUID = 1L;

    public AinosAuthException(String message) {
        super(message);
    }

    public AinosAuthException(String message, Throwable cause) {
        super(message, cause);
    }
}